packages/runtime-common/tests/index-writer-test.ts

```diff
@@ -930,6 +933,195 @@ const tests = Object.freeze({
     );
     assert.strictEqual(entry, undefined, 'deleted modules return undefined');
   },
+
+  'update realm meta when indexing is done': async (
+    assert,
+    { indexWriter, adapter },
+  ) => {
+    await setupIndex(
+      adapter,
+      [{ realm_url: testRealmURL, current_version: 1 }],
+      [
+        {
+          url: `${testRealmURL}1.json`,
+          realm_version: 1,
+          realm_url: testRealmURL,
+          pristine_doc: {
+            id: `${testRealmURL}1.json`,
+            type: 'card',
+            attributes: {
+              name: 'Mango',
+            },
+            meta: {
+              adoptsFrom: {
+                module: `./person`,
+                name: 'Person',
+              },
+            },
+          } as LooseCardResource,
+          search_doc: { name: 'Mango' },
+          deps: [`${testRealmURL}person`],
+          types: [{ module: `./person`, name: 'Person' }, baseCardRef].map(
+            (i) => internalKeyFor(i, new URL(testRealmURL)),
+          ),
+        },
+      ],
+    );
+
+    let resource: CardResource = {
+      id: `${testRealmURL}1.json`,
+      type: 'card',
+      attributes: {
+        name: 'Van Gogh',
+      },
+      meta: {
+        adoptsFrom: {
+          module: `./fancy-person`,
+          name: 'FancyPerson',
+        },
+      },
+    };
+    let batch = await indexWriter.createBatch(new URL(testRealmURL));
+    await batch.invalidate(new URL(`${testRealmURL}1.json`));
+    await batch.updateEntry(new URL(`${testRealmURL}1.json`), {
+      type: 'instance',
+      resource,
+      source: JSON.stringify(resource),
+      lastModified: Date.now(),
+      searchData: { name: 'Van Gogh' },
+      deps: new Set([`${testRealmURL}fancy-person`]),
+      displayNames: ['Fancy Person', 'Person', 'Card'],
+      types: [
+        { module: `./fancy-person`, name: 'FancyPerson' },
+        { module: `./person`, name: 'Person' },
+        baseCardRef,
+      ].map((i) => internalKeyFor(i, new URL(testRealmURL))),
+    });
```

Comment:
this test is werid--we would never change the type of an instance that already exists in the index. this particular instance was originally a Person card, and you changed it to a FancyPerson card. This is actually not something we would permit. A better test would be to just add a new FancyPerson instance instead of modifying the type of an instance that already exists